arkouda/plotting.py

```diff
@@ -50,3 +57,97 @@ def plot_dist(b, h, log=True, xlabel=None, newfig=True):
     if xlabel is not None:
         plt.gca().set_xlabel(xlabel, fontsize=14)
+
+
+def hist_all(ak_df: DataFrame, cols: list = [], log: bool = True):
+    """
+    Create a grid plot histogramming all numeric columns in ak dataframe
+
+    Parameters
+    ----------
+    ak_df : ak.DataFrame
+        Full Arkouda DataFrame containing data to be visualized
+    cols : list
+        (Optional) A specified list of columns to be plotted
+    log : bool
+        use log to scale y
+
+    Notes
+    -----
+    This function displays the plot.
+
+    Examples
+    --------
+    >>> import arkouda as ak
+    >>> from arkouda.plotting import hist_all
+    >>> ak_df = ak.DataFrame({"a": ak.array(np.random.randn(100)),
+                              "b": ak.array(np.random.randn(100)),
+                              "c": ak.array(np.random.randn(100)),
+                              "d": ak.array(np.random.randn(100))
+                              })
+    >>> hist_all(ak_df)
+    """
+
+    if len(cols) == 0:
+        cols = ak_df.columns
+
+    num_rows = int(math.ceil(len(cols) ** 0.5))
+    num_cols = (len(cols) + num_rows - 1) // num_rows
+    fig, axes = plt.subplots(num_rows, num_cols, figsize=(10, 10))
+    fig.tight_layout(pad=2.0)
+
+    if num_rows > 1:
+        axes = axes.flatten()
+    else:
+        axes = [axes]
+
+    for col in cols:
+        try:
+            ax = axes[cols.index(col)]
+            x = ak_df[col]
+
+            if x.dtype == "float64":
+                x = x[~isnan(x)]
+
+            n = len(x)
+            g1 = skew(x)
+
+        except ValueError:
+            GB_df = GroupBy(ak_df[col])
+            new_labels = arange(GB_df.unique_keys.size)
+            newcol = GB_df.broadcast(new_labels)
+            x = newcol[:ak_df.size]
+
+            if x.dtype == "float64":
+                x = x[~isnan(x)]
+
+            n = len(x)
+            g1 = skew(x)
+
+        sigma_g1 = math.sqrt(6 * (n - 2) / ((n + 1) * (n + 3)))
+        # Doane's Formula
+        num_bins = int(1 + math.log2(n) + math.log2(1 + abs(g1) / sigma_g1))
+
+        # Compute histogram counts in arkouda
+        h = histogram(x, num_bins)
+        # Compute bins in numpy
+        if isinstance(x, Datetime):
+            # Matplotlib has trouble plotting np.datetime64 and np.timedelta64
+            bins = (
+                date_range(x.min(), x.max(), periods=num_bins)
+                .to_ndarray()
+                .astype("int")
+            )
+        elif isinstance(x, Timedelta):
+            bins = (
+                timedelta_range(x.min(), x.max(), periods=num_bins)
+                .to_ndarray()
+                .astype("int")
+            )
+        else:
+            bins = np.linspace(x.min(), x.max(), num_bins + 1)[:-1]
+
+        ax.bar(bins, h[1].to_ndarray(), width=bins[1] - bins[0])
+        ax.set_title(col, size=8)
+        if x.max() > 100 * x.min():
+            ax.set_yscale("log")
```

Comment:
I might be misreading but it seems like the `log` bool argument is never actually used. It looks the scale being set to log is trigged by this chunk of logic instead

Reply:
@pierce314159 good call, it's a holdover from testing. I'm removing it so the yscale is automatically determined for every column.
If a user wants the ability to modify the scale moving forward we can think about adding it back in.